### PR TITLE
Rewrite joins written with the USING construct

### PR DESCRIPTION
**Pull request by systay**
If a join has been expressed as:

```
FROM tblA JOIN tblB USING (col1,col2)
```

it will be rewritten to

```
FROM tblA JOIN tblB ON tblA.col1 = tblB.col1 AND tblA.col2 = tblB.col2
```

This allows our planner to recognize these queries and plan them correctly.

### CODE REVIEW

**Comment by derekperkins** [comment]
Will this collapse the `USING` columns on the way out? AFAIK, that's the only functional difference between these two joins.

**Comment by sougou** [comment]
I vaguely remember that there was a specific corner case where this approach wouldn't work. I'll need to dig into some more info to find out why.

**Comment by sougou** [comment]
Ok. I remember now. It becomes ambiguous in the case of multi-table joins (if we don't know the schema). We'll need to lookup the documentation.

**Comment by systay** [comment]
> Ok. I remember now. It becomes ambiguous in the case of multi-table joins (if we don't know the schema)

The differences are all around which columns are available in the result set. 

Here are two examples that show differences that we would need to handle:

```
> select a from t as t1 join t as t2 on t1.a = t2.a limit 1;
(1052, "Column 'a' in field list is ambiguous")

> select a from t as t1 join t as t2 using (a) limit 1;
+---+
| a |
+---+
| 0 |
+---+
1 row in set
Time: 0.016s
```

```
> select * from t as t1 join t as t2 using (a) limit 1;
+---+
| a |
+---+
| 0 |
+---+
1 row in set
Time: 0.013s
> select * from t as t1 join t as t2 on t1.a = t2.a limit 1;
+---+---+
| a | a |
+---+---+
| 0 | 0 |
+---+---+
1 row in set
Time: 0.008s
```

TL;DR; Changing the JOIN also changes the output, which I did not expect. This rewrite is not as easy as I first thought. Backing off for now. :(

File: go/vt/vtgate/planbuilder/builder.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func Build(query string, vschema ContextVSchema) (*engine.Plan, error) {
 	if err != nil {
 		return nil, err
 	}
-	result, err := sqlparser.RewriteAST(stmt)
+	result, err := sqlparser.PrepareAST(stmt, map[string]*querypb.BindVariable{}, "", false)
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/builder.go:63]: we can keep the call to RewriteAST as parameterize is kept false on PrepareAST call

File: go/vt/vtgate/planbuilder/testdata/from_cases.txt [hunk 2]
```diff
@@ -2079,6 +2079,43 @@
   }
 }
 
+# join with USING construct
+"select user.id from user join user_extra using(id)"
+{
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/testdata/from_cases.txt:2084]: add a test for non-vindex column in projection to see if that works

File: go/vt/vtgate/planbuilder/builder.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vtgate/planbuilder/builder.go]: changed in the latest commit.

File: go/vt/vtgate/planbuilder/testdata/from_cases.txt (hunk 2 above)
Comment (systay) [reply go/vt/vtgate/planbuilder/testdata/from_cases.txt:2084]: we're not touching this logic at all. all we are doing is rewriting USING to ON in the rewriter

**Comment by GuptaManan100** [comment]
Fixes #7119

**Review by harshit-gangal — APPROVED** [review]

**Comment by systay** [comment]
Very limited support for using `JOIN ... USING`. 

Limitations:
* `SELECT *` is already not supported in for scatter joins, but double not supported for this form of join syntax.
* No more than two tables can be joined this way.

Some frameworks seem to need at least this form (#7119), and this seemed like an acceptable path forward in lieu of more substantial reworking of the join planning.
